quickjs/src/jniMain/kotlin/app/cash/quickjs/ziplineJni.kt

```diff
@@ -50,25 +51,34 @@ actual abstract class Zipline {
   @PublishedApi
   internal abstract fun set(name: String, service: InboundService<*>)
 
+  /**
+   * Release resources held by this instance. It is an error to do any of the following after
+   * calling close:
+   *
+   *  * Call [get] or [set].
+   *  * Accessing [quickJs].
+   *  * Accessing the objects returned from [get].
+   */
+  abstract override fun close()
+
   companion object {
-    fun create(
-      dispatcher: CoroutineDispatcher,
-    ): Zipline {
+    fun create(dispatcher: CoroutineDispatcher): Zipline {
       val quickJs = QuickJs.create()
       // TODO(jwilson): figure out a 512 KiB limit caused intermittent stack overflow failures.
       quickJs.maxStackSize = 0L
-      return ZiplineJvm(quickJs, dispatcher)
+      return ZiplineJni(quickJs, dispatcher)
     }
   }
 }
 
-private class ZiplineJvm(
+private class ZiplineJni(
```

Comment:
I think this is actuall a `QuickJsZipline`. With a small tweak to extract the call to the logger, we can reuse the entirety of this implementation on Kotlin/Native.